server/src/Main.hs

```diff
@@ -68,28 +67,60 @@ setAuthenticatedAs db userHandle = do
     "Set-Cookie"
     (LText.decodeUtf8 (Builder.toLazyByteString (Cookie.renderSetCookie setCookie)))
 
-getAuthenticatedUser :: Database.Connection -> Scotty.ActionM (Maybe M.UserHandle)
-getAuthenticatedUser db = runMaybeT $ do
+getAuthToken :: MaybeT Scotty.ActionM Database.AuthToken
+getAuthToken = do
   cookieHeader <- MaybeT $ Scotty.header "cookie"
   let cookies = Cookie.parseCookies $ LBS.toStrict $ LText.encodeUtf8 cookieHeader
   sessionCookie <- MaybeT . pure $ lookup "auth-token" cookies
-  token <- MaybeT . pure $ either (const Nothing) (Just . Database.AuthToken) $ Base64.decodeUnpadded sessionCookie
+  MaybeT . pure $ either (const Nothing) (Just . Database.AuthToken) $ Base64.decodeUnpadded sessionCookie
+
+getAuthenticatedUser :: Database.Connection -> Scotty.ActionM (Maybe M.UserAccountName)
+getAuthenticatedUser db = runMaybeT $ do
+  token <- getAuthToken
   MaybeT $
-    liftIO $
+    Scotty.liftAndCatchIO $
       Database.withTransaction db $ \tx ->
         Database.getAuthTokenUser tx token
 
+logout :: Database.Connection -> Scotty.ActionM ()
```

Comment:
Is it perhaps possible to remove the cookie, instead of setting it to ""?

Reply:
The standard way to expire a cookie is by setting `Max-Age` to 0 (or `Expires`), see https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#attributes. So the value that is set doesn't actually matter, it just has to be anything to "override" the previous cookie

Though you do point out to me that the auth token cookie actually doesn't have `Expires` specified, which according to above spec makes it default to a session cookie, which can get cleaned when the browser restarts. I'll change this to be never expiring

Reply:
I've now changed the code to set a `Max-Age` of 1 hour on the auth token cookie, meaning that the user stays logged in for an hour (in practice this should be more like 1 month or so), even if the browser session is closed. It also now refreshes this timer whenever the user is logged in.

Reply:
TIL. :+1: